[hue] Add support for groups/rooms

lolodomo opened the pull request:
Fix #7419

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

lolodomo commented:
@regnets : here is jar in case you would like to test.
[org.openhab.binding.hue-2.5.5-SNAPSHOT.zip](https://github.com/openhab/openhab-addons/files/4533898/org.openhab.binding.hue-2.5.5-SNAPSHOT.zip)

cpmeister commented on bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueGroupHandler.java:
```diff
@@ -0,0 +1,207 @@
+
+    private @NonNullByDefault({}) String groupId;
+
+    private final Logger logger = LoggerFactory.getLogger(HueGroupHandler.class);
```
please move final fields above non-final fields

lolodomo commented on bundles/org.openhab.binding.hue/src/main/resources/ESH-INF/thing/Group.xml:
```diff
@@ -0,0 +1,29 @@
+		<label>Hue Group</label>
+		<description>A group of lights or a room that could be switched on and off.</description>
+
+		<channels>
```
I did not see that until today but yes the hue app allows setting the brightness of a group. And the brightness of a group seems to be the average brightness of its on lights . I will add the brightness control.
For the color control, I don't know yet if this is possible.

regnets replied:
I added a comment on the issue: https://github.com/openhab/openhab-addons/pull/7476
 Based on the unofficial documentation you can send any parameter to a group as you would do with a single light: 

http://www.burgestrand.se/hue-api/api/groups/

http://www.burgestrand.se/hue-api/api/lights/#changing-light-color-and-turning-them-onoff

Maybe give it a try?

lolodomo replied:
I have now added brightness control in addition to switch control.

DerOetzi replied:
yes I think the group can really handle all of the light commands as well

lolodomo replied:
I have now added control control.

lolodomo replied:
I have now added color temperature control.

lolodomo commented:
New jar for testing including switch and brightness controls:
[org.openhab.binding.hue-2.5.5-SNAPSHOT.zip](https://github.com/openhab/openhab-addons/files/4535143/org.openhab.binding.hue-2.5.5-SNAPSHOT.zip)

lolodomo commented:
New jar for testing including switch, brightness and color controls:
[org.openhab.binding.hue-2.5.5-SNAPSHOT.zip](https://github.com/openhab/openhab-addons/files/4536615/org.openhab.binding.hue-2.5.5-SNAPSHOT.zip)

regnets commented:
Ok just tested the binding.

After some fiddling arround, this works great! Its so much cleaner to just turn on/turn off a room instead of turning 13 lights in serial on/off, this is so much more responsible!

I will attach my configuration if someone is also curios about that binding:
Things:
```
Bridge hue:bridge:00178827902c "Bridge: Philips Hue" [ ipAddress="philips-hue", userName="" ] {
 group Hue_Gruppe_Kueche "Philips Hue: Gruppe Kueche" [ groupId="6" ]
}

```

Items:
```
Switch Hue_Gruppe_Kueche_Schalter "Gruppe Küche" <wallswitch>  (Alle_Items) { channel="hue:group:00178827902c:Hue_Gruppe_Kueche:switch" }
Dimmer Hue_Gruppe_Kueche_Dimmer   "Gruppe Küche" <slider>      (Alle_Items) { channel="hue:group:00178827902c:Hue_Gruppe_Kueche:brightness" }
Color  Hue_Gruppe_Kueche_Farbe    "Gruppe Küche" <colorlight>  (Alle_Items) { channel="hue:group:00178827902c:Hue_Gruppe_Kueche:color" }

```


Here are my findings:
- Hue Entertainment Areas are recognized as groups/rooms, i think we need to hide these.
- I got a Hue Group 0 which i dont know, as of testing this looks like the hue group for all lights. Maybe it would be a good idea to rename this group for discovery into "Everything" or something similar.
- For the configuration we are currently using "group" as thing type. Is there a zigbee id for groups of things? So that the configuration would be more to philips current standard?
- Besides the color channel, the color temperature channel would be also a great addition.

regnets commented:
I just had a look at the zigbee light link user guide. It seems like groups are clusters of zigbee lights.

So based on the documentation i would propose to use 0004 as group thing type. Have a look at page 23 of that pdf. What do you think about that?

https://www.nxp.com/docs/en/user-guide/JN-UG-3091.pdf

lolodomo commented:
What is a Hue Entertainment Area ? How can I identify it ? Does it contain lights ? How do you create such an area with the hue app ?

Regarding the group 0, I agree it has to be renamed, I propose "All lights".

No problem to use 0004 as thing type.

Color temperature control will be added.

lolodomo commented:
After a Google search, I think you are talking about the usage with a Hue Play HDMI Sync Box.

regnets commented:
A Hue Entertainment Area is used for the Hue Entertainment Sync. You can create an Entertainment Area in the Hue App in Settings "Entertainment Zones". It is used for light synchronization with the TV through the HDMI Sync Box or a Windows App.

Have a look at it:
https://www2.meethue.com/en-us/entertainment/sync-with-pc

There are also a lot of youtube videos showing this functionality.

On how to identify theses groups - i don't know.

regnets commented:
> After a Google search, I think you are talking about the usage with a Hue Play HDMI Sync Box.

Yes!

regnets commented:
@lolodomo If you are interested, there is also a feature request for scene integration #6043. I added a bounty for that request. Maybe this can be done quite fast as you are currently working on that binding.

lolodomo commented:
Looking at the result of the groups requests, it looks like it exists a "type" field. It can take the value ""LightGroup" or "Room". Maybe this type is different for a Hue Entertainment Areas. I have a bridge v1 and the app does not to create such area. I will log the group type so that you can tell me what value you get for Hue Entertainment Areas.

regnets commented:
> Looking at the result of the groups requests, it looks like it exists a "type" field. It can take the value ""LightGroup" or "Room". Maybe this type is different for a Hue Entertainment Areas. I have a bridge v1 and the app does not to create such area. I will log the group type so that you can tell me what value you get for Hue Entertainment Areas.

Here is my output from api/.../groups:
```
{
    "1": {
        "name": "Entertainment Wohnzimmer",
        "lights": [
            "47",
            "40",
            "41"
        ],
        "sensors": [],
        "type": "Entertainment",
        "state": {
            "all_on": false,
            "any_on": false
        },
        "recycle": false,
        "class": "TV",
        "stream": {
            "proxymode": "auto",
            "proxynode": "/lights/47",
            "active": false,
            "owner": null
        },
        "locations": {
            "47": [
                -0.63,
                0.82,
                0.00
            ],
            "40": [
                0.69,
                0.83,
                0.00
            ],
            "41": [
                1.00,
                0.08,
                1.00
            ]
        },
        "action": {
            "on": false,
            "bri": 190,
            "hue": 7170,
            "sat": 225,
            "effect": "none",
            "xy": [
                0.5267,
                0.4133
            ],
            "ct": 500,
            "alert": "none",
            "colormode": "ct"
        }
    },
    "2": {
        "name": "Schlafzimmer Nachttisch",
        "lights": [
            "37",
            "62"
        ],
        "sensors": [],
        "type": "Zone",
        "state": {
            "all_on": false,
            "any_on": false
        },
        "recycle": false,
        "class": "Bedroom",
        "action": {
            "on": false,
            "bri": 254,
            "hue": 40010,
            "sat": 22,
            "effect": "none",
            "xy": [
                0.3612,
                0.3669
            ],
            "ct": 222,
            "alert": "select",
            "colormode": "ct"
        }
    },
    "3": {
        "name": "Badezimmer",
        "lights": [
            "61",
            "25"
        ],
        "sensors": [],
        "type": "Room",
        "state": {
            "all_on": false,
            "any_on": true
        },
        "recycle": false,
        "class": "Bathroom",
        "action": {
            "on": false,
            "bri": 254,
            "alert": "select"
        }
    },
    "4": {
        "name": "Wohnzimmer",
        "lights": [
            "30",
            "10",
            "11",
            "8",
            "9",
            "7",
            "40",
            "41",
            "47"
        ],
        "sensors": [],
        "type": "Room",
        "state": {
            "all_on": false,
            "any_on": false
        },
        "recycle": false,
        "class": "Living room",
        "action": {
            "on": false,
            "bri": 190,
            "hue": 7170,
            "sat": 225,
            "effect": "none",
            "xy": [
                0.5266,
                0.4133
            ],
            "ct": 500,
            "alert": "none",
            "colormode": "ct"
        }
    },
    "5": {
        "name": "Flur",
        "lights": [
            "31",
            "59"
        ],
        "sensors": [],
        "type": "Room",
        "state": {
            "all_on": false,
            "any_on": false
        },
        "recycle": false,
        "class": "Hallway",
        "action": {
            "on": false,
            "bri": 254,
            "ct": 370,
            "alert": "select",
            "colormode": "ct"
        }
    },
    "6": {
        "name": "Kueche",
        "lights": [
            "42",
            "28",
            "22",
            "23",
            "13",
            "39",
            "53",
            "60"
        ],
        "sensors": [],
        "type": "Room",
        "state": {
            "all_on": false,
            "any_on": true
        },
        "recycle": false,
        "class": "Kitchen",
        "action": {
            "on": false,
            "bri": 254,
            "hue": 41491,
            "sat": 78,
            "effect": "none",
            "xy": [
                0.3120,
                0.3280
            ],
            "ct": 153,
            "alert": "select",
            "colormode": "hs"
        }
    },
    "7": {
        "name": "Schlafzimmer",
        "lights": [
            "37",
            "57",
            "62"
        ],
        "sensors": [],
        "type": "Room",
        "state": {
            "all_on": false,
            "any_on": false
        },
        "recycle": false,
        "class": "Bedroom",
        "action": {
            "on": false,
            "bri": 254,
            "hue": 40010,
            "sat": 22,
            "effect": "none",
            "xy": [
                0.3612,
                0.3669
            ],
            "ct": 222,
            "alert": "select",
            "colormode": "ct"
        }
    },
    "8": {
        "name": "Custom group for $lights",
        "lights": [
            "25"
        ],
        "sensors": [],
        "type": "LightGroup",
        "state": {
            "all_on": false,
            "any_on": false
        },
        "recycle": true,
        "action": {
            "on": false,
            "alert": "select"
        }
    },
    "9": {
        "name": "Allgemein",
        "lights": [
            "26"
        ],
        "sensors": [],
        "type": "Room",
        "state": {
            "all_on": false,
            "any_on": false
        },
        "recycle": false,
        "class": "Other",
        "action": {
            "on": false,
            "alert": "select"
        }
    },
    "10": {
        "name": "Wohnzimmer Decke",
        "lights": [
            "10",
            "11",
            "8",
            "9",
            "7"
        ],
        "sensors": [],
        "type": "Zone",
        "state": {
            "all_on": false,
            "any_on": false
        },
        "recycle": false,
        "class": "Living room",
        "action": {
            "on": false,
            "bri": 2,
            "alert": "select"
        }
    }
}
```

So i can see basically three different group types: Entertainment, Zone and Room. I am not sure what my group 8 is :D.

regnets commented:
Thinking a bit longer about Group 8:
```
    "8": {
        "name": "Custom group for $lights",
        "lights": [
            "25"
        ],
        "sensors": [],
        "type": "LightGroup",
        "state": {
            "all_on": false,
            "any_on": false
        },
        "recycle": true,
        "action": {
            "on": false,
            "alert": "select"
        }
    },
```

I think this is a hidden Group created by the hue bridge to control a single lamp with the Hue Dimmer Switch, as Light Number 25 is the Heater in my Bathroom, so we should also need to hide the type Lightgroup.

DerOetzi commented:
> 
> 
> I just had a look at the zigbee light link user guide. It seems like groups are clusters of zigbee lights.
> 
> So based on the documentation i would propose to use 0004 as group thing type. Have a look at page 23 of that pdf. What do you think about that?
> 
> https://www.nxp.com/docs/en/user-guide/JN-UG-3091.pdf

Can anyone explain me why we are using always the zigbee id for thing-types. I think this is really a pain to use not speaking names for things but cryptic numbers for both developers of binding but above all for users of the binding. I would prefer it to have speaking things name like: dimmable, colorlight, presencesensor, group etc. I know this would be a breaking change for the binding, but in my opinion, would make life for all of us easier in the long run. So maybe we can leave group instead of 0004 and make a new pull request for changing other think names?

regnets commented:
> > I just had a look at the zigbee light link user guide. It seems like groups are clusters of zigbee lights.
> > So based on the documentation i would propose to use 0004 as group thing type. Have a look at page 23 of that pdf. What do you think about that?
> > https://www.nxp.com/docs/en/user-guide/JN-UG-3091.pdf
> 
> Can anyone explain me why we are using always the zigbee id for thing-types. I think this is really a pain to use not speaking names for things but cryptic numbers for both developers of binding but above all for users of the binding. I would prefer it to have speaking things name like: dimmable, colorlight, presencesensor, group etc. I know this would be a breaking change for the binding, but in my opinion, would make life for all of us easier in the long run. So maybe we can leave group instead of 0004 and make a new pull request for changing other thing names?

I totally agree with you. I just made the proposal to use 0004 in order to use the same standard. Maybe this is a thing which can be discussed with @kaikreuzer ? I would personally also prefer more memorable names.

lolodomo commented:
Here is a new version with the following changes:

- Optimized way to retrieve groups information
- Hue Entertainment areas ignored by the discovery process
- Group 0 labelled "All lights" by the discovery process
- Group type added to the label by the discovery process

[org.openhab.binding.hue-2.5.5-SNAPSHOT.zip](https://github.com/openhab/openhab-addons/files/4539292/org.openhab.binding.hue-2.5.5-SNAPSHOT.zip)

DerOetzi commented:
> 
> 
> Thinking a bit longer about Group 8:
> 
> ```
>     "8": {
>         "name": "Custom group for $lights",
>         "lights": [
>             "25"
>         ],
>         "sensors": [],
>         "type": "LightGroup",
>         "state": {
>             "all_on": false,
>             "any_on": false
>         },
>         "recycle": true,
>         "action": {
>             "on": false,
>             "alert": "select"
>         }
>     },
> ```
> 
> I think this is a hidden Group created by the hue bridge to control a single lamp with the Hue Dimmer Switch, as Light Number 25 is the Heater in my Bathroom, so we should also need to hide the type Lightgroup.

I don't think it's a good thing to hide Lightgroup, this is the basic group type for lights or plugs. Many diy hue bridges for example deconz use only groups of this type

lolodomo commented:
I don"r hide the groups of type "LightGroup"; I only hide the ones of type "Entertainment".

regnets commented:
Ok, here is how my inbox now looks like:
![image](https://user-images.githubusercontent.com/1339599/80368411-ff679980-888c-11ea-912f-7c5efcd60710.png)

I would prefer the group type in the ?detailed description?, this would be also suitable for the release version. I made a mock up for this:
![image](https://user-images.githubusercontent.com/1339599/80368680-638a5d80-888d-11ea-928e-b43c1e448785.png)

As you can see Group 0 is not renamed to "All lights", maybe this is an issue with my cache or temporary folders? I just removed the old jar waited a few minutes and copied the new jar in the addon folder.

I like @DerOetzi idea about renaming the thing types for that binding. I am curios if we could do something like a downwards compatible version. To accept both the Zigbee Id and from now on the new thing types.

lolodomo commented:
> @lolodomo If you are interested, there is also a feature request for scene integration #6043. I added a bounty for that request. Maybe this can be done quite fast as you are currently working on that binding.

There was PR 6044 for that, no ?
But the Git link to this PR is broken ! Maybe the user deleted his PR.

regnets commented:
> > @lolodomo If you are interested, there is also a feature request for scene integration #6043. I added a bounty for that request. Maybe this can be done quite fast as you are currently working on that binding.
> 
> There was PR 6044 for that, no ?
> But the Git link to this PR is broken ! Maybe the user deleted his PR.

Yes, thats why I asked you :). Maybe this could also be added to rooms as scenes are always related to groups (rooms/zones).

lolodomo commented:
> 
> 
> > > @lolodomo If you are interested, there is also a feature request for scene integration #6043. I added a bounty for that request. Maybe this can be done quite fast as you are currently working on that binding.
> > 
> > 
> > There was PR 6044 for that, no ?
> > But the Git link to this PR is broken ! Maybe the user deleted his PR.
> 
> Yes, thats why I asked you :). Maybe this could also be added to rooms as scenes are always related to groups (rooms/zones).

I will first wait for the merge of this PR.

lolodomo commented:
> As you can see Group 0 is not renamed to "All lights", maybe this is an issue with my cache or temporary folders? I just removed the old jar waited a few minutes and copied the new jar in the addon folder.

Group 0 is added manually because with old bridges, it was not returned by the "groups" request. Probably with your more recent bridge or firmware, this "bug" was fixed by Philips. So I have a little change to do to not request this group if already provided by the "groups" request and for the renaming in the discovery, I have to consider its id and not its name.

DerOetzi commented:
> 
> 
> Here is a new version with the following changes:
> 
>     * Optimized way to retrieve groups information
> 
>     * Hue Entertainment areas ignored by the discovery process
> 
>     * Group 0 labelled "All lights" by the discovery process
> 
>     * Group type added to the label by the discovery process
> 
>     * Thing type for group is now 0004
> 
> 
> [org.openhab.binding.hue-2.5.5-SNAPSHOT.zip](https://github.com/openhab/openhab-addons/files/4539292/org.openhab.binding.hue-2.5.5-SNAPSHOT.zip)

Not only about the speaking name argument above, but I think with using 0004 ClusterID for groups, we start to mix up two different layers of ZigBee device layer and cluster layer.

lolodomo commented:
New version for renaming group 0:
[org.openhab.binding.hue-2.5.5-SNAPSHOT.zip](https://github.com/openhab/openhab-addons/files/4539467/org.openhab.binding.hue-2.5.5-SNAPSHOT.zip)


> Not only about the speaking name argument above, but I think with using 0004 ClusterID for groups, we start to mix up two different layers of ZigBee device layer and cluster layer.

This was your idea ! I can revert.

lolodomo commented:
> I would prefer the group type in the ?detailed description?, this would be also suitable for the release version. I made a mock up for this

This is not possible. Paper UI displays the thing type label at this place.

regnets commented:
> New version for renaming group 0:
> [org.openhab.binding.hue-2.5.5-SNAPSHOT.zip](https://github.com/openhab/openhab-addons/files/4539467/org.openhab.binding.hue-2.5.5-SNAPSHOT.zip)
> 
> > Not only about the speaking name argument above, but I think with using 0004 ClusterID for groups, we start to mix up two different layers of ZigBee device layer and cluster layer.
> 
> This was your idea ! I can revert.

You are absolutely right, that is my idea. 

I just wanted to stay to the current notation. However, I don't like the notation myself.

If we can decide that i would prefer for all thing more memorable, self descriptive names.

lolodomo commented:
New version with thing type reverted to "group":
[org.openhab.binding.hue-2.5.5-SNAPSHOT.zip](https://github.com/openhab/openhab-addons/files/4539499/org.openhab.binding.hue-2.5.5-SNAPSHOT.zip)

lolodomo commented:
I am not happy with the introduced conflict !

regnets commented:
> I am not happy with the introduced conflict !

If i am responsible for this: Sorry!

lolodomo commented:
> > I am not happy with the introduced conflict !
> 
> If i am responsible for this: Sorry!

No, of course not you. There was probably a merged PR that introduced this conflict. As I see no recent merge about the hue binding, this is probably a generic PR, and this is not cool.

lolodomo commented:
Renaming thing types will not be part of this PR. I don't like this idea because it will break the setup of all users using this binding !

regnets commented:
> Renaming thing types will not be part of this PR. I don't like this idea because it will break the setup of all users using this binding !

I will create a new issue for that.

lolodomo commented:
Regarding the group type added to the discovery label, this was just an idea to show this type. But I can remove it.

regnets commented:
> Regarding the group type added to the discovery label, this was just an idea to show this type. But I can remove it.

I find it quite useful, from my point of view this can be also in the final version. If you don't want the type, just rename it. Otherwise you won't have that information in the paper ui.

lolodomo commented:
So let me know if the last version is OK.
I still have to add the color temperature control.

DerOetzi commented:
> 
> 
> Renaming thing types will not be part of this PR. I don't like this idea because it will break the setup of all users using this binding !

Totally agree: That shouldn't be part of this PR because it should be discussed thoroughly for being a breaking change. Although I basically support the idea of descriptive thing names

regnets commented:
> So let me know if the last version is OK.
> I still have to add the color temperature control.

I had to clean my cache and my tmp directory, after that everything looks fine and works as expected. Also the Group 0 is now named All Lights.

lolodomo commented:
Color temperature control is now implemented.
I rebased the PR to solve the conflict.
The enhancement is finished in my point of view.
Here is new jar for testing:
[org.openhab.binding.hue-2.5.5-SNAPSHOT.zip](https://github.com/openhab/openhab-addons/files/4541749/org.openhab.binding.hue-2.5.5-SNAPSHOT.zip)

lolodomo commented:
I have a doubt if this is necessary to have 4 channels on a group thing: switch, brigthness, color and color_temperature.

lolodomo commented:
@cpmeister : could you please have a final look ?
@regnets confirmed that it works well.

cpmeister commented on bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java:
```diff
@@ -765,6 +910,36 @@ private void notifySensorStatusListeners(final FullSensor fullSensor, final Stri
+                                "Could not notify groupStatusListeners for unknown event type " + type);
+                }
+            } catch (Exception e) {
+                logger.error("An exception occurred while calling the Group Listeners", e);
```
The error logging level should be reserved for issues catastrophic enough to threaten the operation of openHab itself. An error in a binding shouldn't qualify for this. Please read the [openHab coding guidelines](https://www.openhab.org/docs/developer/guidelines.html#f-logging) for expected use of logging in bindings.

My general rule for logging levels is:
* trace - used for normal expected execution paths
* debug - used for unexepected execution paths, but well within normal operation
* info - used for notable points in an execution path, like a milestone. (in openhab we try reserve this logging level for the core, so bindings should rarely ever call this.)
* warn - used for notable unexpected execution paths that a regular user (not just a developer) should be notified of. Warnings should be used to indicate that something not-normal occurred and user intervention is required to resolve. Warnings do not indicate a failure to operate merely an abnormal condition of operation that can still be handled by the binding. Failures in binding operation should be indicated by changing the thing status to offline.
* error - used to indicate catastrophic program failure. This should be used to indicate a catastrophic failure in openhab's ability to operate. A failure in a binding would never cause openhab as a whole to fail so a failure in a bindings should never log an error. Instead that failure should be indicated by changing the thing status.

lolodomo replied:
Ok, updated.

cpmeister commented on bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueGroupHandler.java:
```diff
@@ -0,0 +1,373 @@
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        HueClient handler = getHueClient();
+        if (handler != null) {
+            FullGroup group = handler.getGroupById(groupId);
+            if (group != null) {
+                onGroupStateChanged(null, group);
+            }
+        }
```
When this binding starts up this will get called for every channel. If you are just updating the channels with cached data it should be fine, but if you are making network requests or the like then you should use an expiring cache.

lolodomo replied:
It is just using the cache:
```
    public @Nullable FullGroup getGroupById(String groupId) {
        return lastGroupStates.get(groupId);
    }
```

cpmeister commented on bundles/org.openhab.binding.hue/src/main/resources/ESH-INF/thing/Group.xml:
```diff
@@ -0,0 +1,37 @@
+				<description>The group identifier identifies one certain hue group or room.</description>
+				<required>true</required>
+			</parameter>
+			<parameter name="fadetime" type="integer" min="0" step="100">
```
```suggestion
			<parameter name="fadetime" type="integer" min="0" step="100" unit="ms">
```

lolodomo replied:
Ok, done.
Another PR would be welcome to fix all other thing types.

cpmeister commented on bundles/org.openhab.binding.hue/src/main/resources/ESH-INF/thing/Group.xml:
```diff
@@ -0,0 +1,37 @@
+			</parameter>
+			<parameter name="fadetime" type="integer" min="0" step="100">
+				<label>Fade Time</label>
+				<description>Fade time in ms for changing values</description>
```
```suggestion
				<description>Fade time for changing values</description>
```

lolodomo replied:
Changed.

lolodomo replied:
I revert this one becauyse Paper UI is not displaying th eunit, so without this information in the description, the user will not know what is the unit of this configuration setting.

lolodomo replied:
For your information: https://github.com/eclipse/smarthome/issues/3974

cweitkamp replied:
I know that some UIs (e.g. Paper UI) - which will be removed in OH3 - do not show this "unit" but we prefer to not have it inside the description.

lolodomo replied:
Without the unit in the description, the user will not know how to adjust the setting.
Hopefully, the unit is inside the description in every existing bindings.
I would suggest to change thjat when OH3 is released, not before. And this will require to change more or less every bindings.

lolodomo commented:
Ok, I have now rebased the PR to solve the conflict.

cpmeister commented:
Manually check sign-off
